upstream.sh

```diff
@@ -11,6 +11,10 @@ PROJECT_NAME=$(basename `git rev-parse --show-toplevel`)
 echo $PROJECT_NAME
 COVERAGE_SHA=$(git rev-list HEAD | head -1)
 echo $COVERAGE_SHA > SHA.txt
+echo "Processing coverage for report belonging to latest commit"
+PROD_COVERAGE=$(curl https://codecov.io/api/gh/codecov/$PROJECT_NAME/commit/$COVERAGE_SHA | \
+python2 -c 'import json,sys;obj=json.load(sys.stdin);print obj["commit"]["totals"]["c"]')
+echo $PROD_COVERAGE > coverage_totals.txt
```

Comment:
Turning a bit of a security eye to this approach, you're pulling in an unverified result from an API and storing that into a file to be used later. I think it's necessary to validate the input you're passing into coverage_totals.txt and ensuring it's a number, then failing the result if it isn't. I'd validate `$PROD_COVERAGE` before storing it. 

If you want to keep using python because now you're parsing JSON _and_ validating input, you can. But a simple regex could likely also accomplish the validation if you want to try and remove the python dependency from this script.

Reply:
Agreed, did the validation with a regex after curling the value